lib/components/user/favorite-locations-pane.js

```diff
@@ -37,135 +37,111 @@ const NewLocationFormControl = styled(FormControl)`
 `
 
 // Helper filter functions.
-const isHome = loc => loc.type === 'home'
-const isWork = loc => loc.type === 'work'
-const notHomeOrWork = loc => loc.type !== 'home' && loc.type !== 'work'
+export const isHome = loc => loc.type === 'home'
+export const isWork = loc => loc.type === 'work'
 
 /**
- * User's saved locations editor.
+ * Helper function that adds a new address to the Formik state
+ * using the Formik-provided arrayHelpers object.
  */
-class FavoriteLocationsPane extends Component {
-  static propTypes = {
-    onUserDataChange: PropTypes.func.isRequired,
-    userData: PropTypes.object.isRequired
+function addNewAddress (arrayHelpers, e) {
+  const value = (e.target.value || '').trim()
+  if (value.length > 0) {
+    arrayHelpers.push({
+      address: value,
+      icon: 'map-marker',
+      type: 'custom'
+    })
+
+    // Empty the input box value so the user can enter their next location.
+    e.target.value = ''
   }
+}
 
-  _handleAddNewLocation = e => {
-    const value = e.target.value || ''
-    if (value.trim().length > 0) {
-      const { userData, onUserDataChange } = this.props
-      // FIXME: remove assigning [] when null.
-      const { savedLocations = [] } = userData
-
-      // Create a copy of savedLocations and add the new location to the copied array.
-      const newLocations = clone(savedLocations)
-      newLocations.push({
-        address: value.trim(),
-        icon: 'map-marker',
-        type: 'custom'
-      })
-
-      // Event onChange will trigger after this and before rerender,
-      // so DO empty the input box value so the user can enter their next location.
-      e.target.value = null
-
-      onUserDataChange({ savedLocations: newLocations })
+/**
+ * User's saved locations editor.
+ * TODO: Discuss and improve handling of location details (type, coordinates...).
+ */
+class FavoriteLocationsPane extends Component {
+  _handleNewAddressKeyDown = memoize(
+    arrayHelpers => e => {
+      if (e.keyCode === 13) {
```

Comment:
A raw keycode int should never be undocumented in code.

Reply:
Addressed in https://github.com/opentripplanner/otp-react-redux/pull/239/commits/fb18f75ff1e1417ab6e8a359b5091fba4cdcc6d3